### PR TITLE
assert: handle array case in copyExportedFields #1404

### PR DESCRIPTION
**Pull request by qerdcv**
## Summary
Implemented reflect.Array handling in the copyExportedFields to prevent panic when trying to make a slice from an array

## Changes
Creating new array and filling it with exported values following the example from reflect.Slice.

I've decide to copy paste some code from the reflect.Slice case to make the code more readable, straightforward and avoid mutating the existing case for reflect.Slice

## Motivation
Fixing a bug described in the issue #1404

## Related issues
#1404

### CODE REVIEW

**Comment by qerdcv** [comment]
@glesica 
@boyan-soubachov

Сheck it please, when you have time and inspiration

File: assert/assertions.go [hunk 1]
```diff
@@ -109,7 +109,19 @@ func copyExportedFields(expected interface{}) interface{} {
 		result.Elem().Set(reflect.ValueOf(unexportedRemoved))
 		return result.Interface()
 
-	case reflect.Array, reflect.Slice:
+	case reflect.Array:
+		result := reflect.New(reflect.ArrayOf(expectedValue.Len(), expectedType.Elem())).Elem()
+		for i := 0; i < expectedValue.Len(); i++ {
+			index := expectedValue.Index(i)
+			if isNil(index) {
+				continue
```
Comment (dolmen) [inline assert/assertions.go:117]: This case is not covered by the test suite.

**Review by dolmen — CHANGES_REQUESTED** [review]

File: assert/assertions.go (hunk 1 above)
Comment (qerdcv) [reply assert/assertions.go:117]: Thanks!

I added one more test for nil-value, but it failed. I fixed it by passing an actual `interface{}` to the isNil function.

File: assert/assertions_test.go [hunk 2]
```diff
@@ -402,6 +407,72 @@ func TestEqualExportedValues(t *testing.T) {
...
+	            	+   (int) 3
+	            	   },`,
+		},
+		{
+			value1: S7{
+				Nested{[2]interface{}{nil, 2}, [2]int{1, 2}},
+				Nested{[2]int{1, 2}, [2]int{3, 4}},
+			},
+			value2: S7{
+				Nested{[2]interface{}{nil, 2}, [2]int{3, 4}},
```
Comment (dolmen) [inline assert/assertions_test.go:460]: As `nil` is a special value which means *anything matches*, it would be better to use a concrete value here. Or better: add more test cases to validate that anything matches, not just that `nil` matches *anything*.

**Review by dolmen — CHANGES_REQUESTED** [review]

**Comment by dolmen** [comment]
Issue #1404 has been fixed by #1473.
